soroban-env-host/src/storage.rs

```diff
@@ -476,3 +476,19 @@ impl Storage {
         Ok(())
     }
 }
+
+#[cfg(any(test, feature = "testutils"))]
+impl Storage {
+    /// Unmetered function to access entry with lifetimes to test lifetime extensions.
+    pub fn get_entry_with_lifetime(
+        &self,
+        key: Rc<LedgerKey>,
+    ) -> Result<Option<EntryWithLiveUntil>, HostError> {
+        let entry = self.map.map.iter().find(|e| e.0 == key);
+
+        match entry {
+            None => Err((ScErrorType::Storage, ScErrorCode::InternalError).into()),
```

Comment:
Why is this an error? It seems like you should just return the retrieved `entry` (i.e. `None`)

Reply:
Yeah I copied the logic from try_get_full but since this is just a testutil without prepare_read_only_access a missing key doesn't mean it's an internal error so returning just the option sounds more correct.